### PR TITLE
Change mag calibration math to use matrix library

### PR DESCRIPTION
**Pull request by bugobliterator**

### CODE REVIEW

**Comment by bugobliterator** [comment]
matrix library PR is here: https://github.com/PX4/Matrix/pull/37

File: src/modules/commander/commander.cpp [hunk 1]
```diff
@@ -1702,7 +1702,7 @@ int commander_thread_main(int argc, char *argv[])
 	/* initialize low priority thread */
 	pthread_attr_t commander_low_prio_attr;
 	pthread_attr_init(&commander_low_prio_attr);
-	pthread_attr_setstacksize(&commander_low_prio_attr, PX4_STACK_ADJUSTED(3000));
+	pthread_attr_setstacksize(&commander_low_prio_attr, PX4_STACK_ADJUSTED(5000));
```
Comment (jgoppert) [inline src/modules/commander/commander.cpp]: Is the increased stack still needed?

Comment (bugobliterator) [inline src/modules/commander/commander.cpp]: yes

**Comment by bugobliterator** [comment]
@jgoppert done

**Comment by jgoppert** [comment]
This needs an update for the matrix lib test.:

```bash
../src/systemcmds/tests/test_matrix.cpp:210:29: error: 'matrix::Quatf {aka class matrix::Quaternion<float>}' has no member named 'derivative'

  Vector<float, 4> q_dot = q.derivative(Vector3f(1, 2, 3));
```

In the new library I added 2 functions.

@LorenzMeier any reason we still have test_matrix.cpp? The unit testing in matrix should cover this sort of thing and deleting the redundant testing will require less maintenance.

**Comment by LorenzMeier** [comment]
@jgoppert Not really. I think this could be removed.

**Comment by dagar** [comment]
@jgoppert test_matrix exists so that we can test on the actual platform (anything running PX4).
Is there a way we could slightly restructure the tests in Matrix/test and use them directly?

**Comment by jgoppert** [comment]
The source code exists in the cloned git repo. So we could just include it and run that against the arm compiler.

**Comment by dagar** [comment]
@bugobliterator matrix has been updated in master. Can you rebase?

**Comment by bugobliterator** [comment]
@dagar done

File: src/modules/commander/commander.cpp (hunk 1 above)
Comment (LorenzMeier) [inline src/modules/commander/commander.cpp]: We can only really afford this once this runs in the even library. @bkueng when we've moved this to events, this PR can go in.

**Comment by LorenzMeier** [comment]
Closing as stale.

**Pull request closed without merging**